src/qint.c

```diff
@@ -109,26 +122,45 @@ QINT_API size_t qint_encode4(BufferWriter *bw, uint32_t i1, uint32_t i2, uint32_
   return ret;
 }
 
-#define QINT_DECODE_VALUE(lval, bits, ptr, nused) \
-  do {                                            \
-    switch (bits) {                               \
-      case 2:                                     \
-        lval = *(uint32_t *)ptr & 0xFFFFFF;       \
-        nused = 3;                                \
-        break;                                    \
-      case 0:                                     \
-        lval = *ptr;                              \
-        nused = 1;                                \
-        break;                                    \
-      case 1:                                     \
-        lval = *(uint32_t *)ptr & 0xFFFF;         \
-        nused = 2;                                \
-        break;                                    \
-      default:                                    \
-        lval = *(uint32_t *)ptr;                  \
-        nused = 4;                                \
-        break;                                    \
-    }                                             \
+/* Encode a pair of (32,64) integer */
+QINT_API size_t qint_encode32_64pair(BufferWriter *bw, uint32_t n32, uint64_t n64) {
```

Comment:
The point of this is to be able to write at least _one_ byte inside the header. This way we can distinguish between the 'double' format (where the bits are empty) and the 'uint64' format. The practical distinction is that the pair `(0x01,0x01)` will leave a pattern of `0000` but `0x01 | (0x01<<31)` will actually write at least one bit there, so we'd be able to distinguish.

In theory this consumes more space, but in reality I would imagine the _actual_ space savings to be minimal